src/common/operator/cast_operators.cpp

```diff
@@ -1601,8 +1602,14 @@ struct DecimalCastOperation {
 		}
 		state.digit_count++;
 		if (NEGATIVE) {
+			if (state.result < (NumericLimits<typename T::type_t>::Minimum() / 10)) {
```

Comment:
nit: this could just be `NumericLimits<T>`, and we could remove the `type_t`, no?

Reply:
Ah there are different T's present here, within the scope of the handledigit function, T is the type of state (DecimalCastData) not the numeric type

Reply:
Oh, I see, that is a bit confusing indeed. Perhaps instead rename one of the T's to remove that ambiguity?

Reply:
I've altered it!